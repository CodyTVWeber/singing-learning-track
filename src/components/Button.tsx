import React from 'react';
import { colors, spacing, borderRadius, fontSize, fontWeight, shadows, gradients } from '../theme/theme';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary' | 'outline' | 'ghost' | 'gradient';
  size?: 'small' | 'medium' | 'large';
  fullWidth?: boolean;
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  children: React.ReactNode;
}

export const Button: React.FC<ButtonProps> = ({ 
  variant = 'primary', 
  size = 'medium',
  fullWidth = false,
  icon,
  iconPosition = 'left',
  children,
  style,
  ...props 
}) => {
  const baseStyles: React.CSSProperties = {
    padding: size === 'small' ? `${spacing.sm} ${spacing.md}` : 
             size === 'large' ? `${spacing.md} ${spacing.xl}` : 
             `${spacing.md} ${spacing.lg}`,
    borderRadius: borderRadius.pill,
    fontSize: size === 'small' ? fontSize.sm : 
              size === 'large' ? fontSize.lg : 
              fontSize.md,
    fontWeight: fontWeight.semibold,
    transition: 'none',
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    gap: spacing.sm,
    width: fullWidth ? '100%' : 'auto',
    minHeight: size === 'small' ? '36px' : 
               size === 'large' ? '56px' : 
               '44px',
    cursor: props.disabled ? 'not-allowed' : 'pointer',
    opacity: props.disabled ? 0.6 : 1,
    position: 'relative',
    overflow: 'hidden',
    border: 'none',
    textDecoration: 'none',
    whiteSpace: 'nowrap',
    userSelect: 'none',
    flexDirection: iconPosition === 'right' ? 'row-reverse' : 'row',
  };

  const variantStyles: Record<string, React.CSSProperties> = {
    primary: {
      backgroundColor: colors.primary,
      color: colors.textOnPrimary,
      boxShadow: shadows.md,
    },
    secondary: {
      backgroundColor: colors.secondary,
      color: colors.textOnSecondary,
      boxShadow: shadows.md,
    },
    outline: {
      backgroundColor: 'transparent',
      color: colors.primary,
      border: `2px solid ${colors.primary}`,
      boxShadow: 'none',
    },
    ghost: {
      backgroundColor: 'transparent',
      color: colors.text,
      boxShadow: 'none',
    },
    gradient: {
      background: gradients.primary,
      color: colors.textOnPrimary,
      boxShadow: shadows.lg,
    },
  };

  // Simplified button: no hover/active animations

  return (
    <button
      {...props}
      style={{
        ...baseStyles,
        ...variantStyles[variant],
        ...style,
      }}
    >
      {icon && <span style={{ display: 'flex', alignItems: 'center' }}>{icon}</span>}
      {children}
    </button>
  );
};