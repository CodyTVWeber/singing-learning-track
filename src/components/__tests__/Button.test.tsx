import React from 'react';
import { render } from '@testing-library/react';
import { Button } from '../Button';

describe('Button', () => {
  it('renders primary medium button snapshot', () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});

