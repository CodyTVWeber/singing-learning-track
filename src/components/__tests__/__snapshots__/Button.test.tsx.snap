// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders primary medium button snapshot 1`] = `
<DocumentFragment>
  <button
    style="padding: 16px 24px; border-radius: 9999px; font-size: 16px; font-weight: 600; transition: none; display: inline-flex; align-items: center; justify-content: center; gap: 8px; width: auto; min-height: 44px; cursor: pointer; opacity: 1; position: relative; overflow: hidden; text-decoration: none; white-space: nowrap; user-select: none; flex-direction: row; background-color: rgb(139, 115, 85); color: rgb(250, 246, 242); box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.07), 0 2px 4px -1px rgba(0, 0, 0, 0.04);"
  >
    Click me
  </button>
</DocumentFragment>
`;
